### ISSUE TITLE
False positive for `react/prop-types` in decorator

### ISSUE THREAD

**Issue by EECOLOR** [yannickcr/eslint-plugin-react#619]
Minimal example:

```
import { Component } from 'react'

const test = null
@test(_ => ({ arg }) => arg)
export class Test extends Component {}
```

It reports the following error:
> 'arg' is missing in props validation (react/prop-types)

**Comment by yannickcr** [comment]
Seems to be the same issue as #591

It should be fixed in the next release, fill free to re-open if it is not the case.
